lib/ansible/modules/cloud/amazon/ecs_attribute.py

```diff
@@ -0,0 +1,251 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ecs_attribute
+short_description: manage ecs attributes
+description:
+    - Create, update, delete or list ECS container instance attributes.
+version_added: "2.3"
+author: Andrej Svenke (@anryko)
+requirements: [ boto, botocore, boto3 ]
+options:
+    cluster:
+        description:
+            - The short name or full Amazon Resource Name (ARN) of the cluster that contains the resource to apply attributes.
+        required: true
+    state:
+        description:
+            - The desired state of the attribute.
+        required: true
+        choices: ['present', 'absent']
+    name:
+        description:
+            - The name of the attribute. Up to 128 letters (uppercase and lowercase), numbers, hyphens, underscores, and periods are allowed.
+        required: true
+    value:
+        description:
+            - The value of the attribute. Up to 128 letters (uppercase and lowercase), numbers, hyphens, underscores, periods, at signs (@),
+              forward slashes, colons, and spaces are allowed.
+        required: false
+    ec2_instance_id:
+        description:
+            - EC2 instance ID of ECS cluster container instance.
+        required: true
+extends_documentation_fragment:
+    - aws
+    - ec2
+'''
+
+EXAMPLES = '''
+# Note: These examples do not set authentication details, see the AWS Guide for details.
+
+# Set attribute
+- ecs_attribute:
+    cluster: test-cluster
+    name: flavor
+    value: test
+    state: present
+    ec2_instance_id: "{{ ec2_id }}"
+  delegate_to: localhost
+
+# Delete attribute
+- ecs_attribute:
+    cluster: test-cluster
+    name: flavor
+    state: absent
+    ec2_instance_id: "{{ ec2_id }}"
+  delegate_to: localhost
+'''
+
+RETURN = '''
+attributes:
+    description: list of attributes
+    type: list of objects
+    contains:
+        name:
+            description: name of the attribute
+            type: string
+        value:
+            description: value of the attribute
+            returned: if present
+            type: string
+        cluster:
+            description: cluster name
+            type: string
+        ec2_instance_id:
+            description: ec2 instance id of ecs container instance
+            type: string
+'''
+
+try:
+    import boto
+    import botocore
```

Comment:
`import botocore` should be in the HAS_BOTO3 check.

I'm not convinced that `boto` is required here - you don't seem to use it at all - can you remove all references to HAS_BOTO if that's the case.